### PR TITLE
A third definition of less-than for Nats

### PR DESCRIPTION
**Pull request by xekoukou**

### CODE REVIEW

**Comment by MatthewDaggitt** [comment]
Well certainly completes the set! Do you have any examples where this is useful?

File: src/Data/Nat/Properties.agda [hunk 1]
```diff
@@ -822,6 +822,21 @@ i^j≡1⇒j≡0∨i≡1 : ∀ i j → i ^ j ≡ 1 → j ≡ 0 ⊎ i ≡ 1
...
+------------------------------------------------------------------------
+-- Properties of _≤‴_
+
+≤‴⇒≤″ : ∀{m n} → m ≤‴ n → m ≤″ n
+≤‴⇒≤″ {m = m} ≤‴-refl = less-than-or-equal {k = 0} (+-identityʳ m)
+≤‴⇒≤″ {m = m} (≤‴-step x) = less-than-or-equal (trans (+-suc m _) (_≤″_.proof ind)) where
+  ind = ≤‴⇒≤″ x
+
+≤″⇒≤‴ : ∀{m n} → m ≤″ n → m ≤‴ n
+≤″⇒≤‴ (less-than-or-equal {k} proof) = h proof where
```
Comment (MatthewDaggitt) [inline src/Data/Nat/Properties.agda]: I think this proof simplifies by pattern matching on `proof` to `refl`?

File: src/Data/Nat/Base.agda [hunk 2]
```diff
@@ -129,6 +129,15 @@ m ≥′ n = n ≤′ m
 _>′_ : Rel ℕ 0ℓ
 m >′ n = n <′ m
 
+------------------------------------------------------------------------
+-- Similar to _≤′_
+
+infix 4 _≤‴_
+
+data _≤‴_ : ℕ → ℕ → Set where
```
Comment (MatthewDaggitt) [inline src/Data/Nat/Base.agda]: Can this be moved below `_≤''_` so they're in the right order numerically.

**Comment by oisdk** [comment]
If you swap the arguments, you can change one of the indices to a parameter:

```agda
infix 4 _≥‴_
data _≥‴_ (m : ℕ) : ℕ → Set where
  m≥m : m ≥‴ m
  m≥p : ∀ {n} → m ≥‴ suc n → m ≥‴ n
```

Other than that, this is the same as the [type I use for modular arithmetic](https://gist.github.com/oisdk/037862acb385f30257a1c594fd5a616e).

File: src/Data/Nat/Properties.agda (hunk 1 above)
Comment (xekoukou) [reply src/Data/Nat/Properties.agda]: I couldn't make it simpler. Do you have something in mind?

Comment (gallais) [reply src/Data/Nat/Properties.agda]: You could do something like this:

```agda
m≤‴n+m : ∀ {m} n → m ≤‴ n + m
m≤‴n+m zero = ≤‴-refl
m≤‴n+m {m} (suc n) rewrite sym (+-suc n m) = ≤‴-step (m≤‴n+m n)

m≤‴m+n : ∀ {m} n → m ≤‴ m + n
m≤‴m+n {m} n rewrite +-comm m n = m≤‴n+m n

≤″⇒≤‴ : ∀{m n} → m ≤″ n → m ≤‴ n
≤″⇒≤‴ {m} (less-than-or-equal {k} refl) = m≤‴m+n k
```

Comment (xekoukou) [reply src/Data/Nat/Properties.agda]: I try not to use ```with``` or substitution in the inductive step. For similar reasons, we have [cast](https://github.com/agda/agda-stdlib/blob/64c90354189b9d497310197cccb989d30caa715e/src/Data/Fin/Base.agda#L47) for ```Fin```.

File: src/Data/Nat/Base.agda (hunk 2 above)
Comment (MatthewDaggitt) [reply src/Data/Nat/Base.agda]: Even so, I think they should still go in order. It'd be good if you could expand the comment as well to explain that it's useful for induction.

**Comment by MatthewDaggitt** [comment]
> I can also swap the arguments if you like.

Hmm I think better to maintain consistency with the other forms.

That said it would be great if you could also add the converse and strict versions as well.

I'll then merge it in :smile:

**Comment by xekoukou** [comment]
@MatthewDaggitt Done.

**Comment by MatthewDaggitt** [comment]
Great, thanks for the PR!

**Pull request merged**